ufficio_soci/tests.py

```diff
@@ -2263,4 +2266,64 @@ def test_cancellazione_quota_socio(self):
+        crea_appartenenza(sostenitore1, sede, tipo=Appartenenza.SOSTENITORE)
+        crea_appartenenza(sostenitore2, sede, tipo=Appartenenza.SOSTENITORE)
+
+        # crea locazione, aspiranti e corso
+        locazione = crea_locazione()
+        Aspirante.objects.create(raggio=900, locazione=locazione, persona=aspirante1)
+        Aspirante.objects.create(raggio=900, locazione=locazione, persona=aspirante2)
+        corso_base = CorsoBase.objects.create(locazione=locazione, stato=Delega.ATTIVA, sede=sede,
+                                              data_inizio=poco_fa(),
+                                              data_esame=datetime.date(2018, 10, 10),
+                                              anno=2018, progressivo=1)
+
+        # nomina delegato corso
+        ctype = ContentType.objects.get_for_model(CorsoBase)
+        Delega.objects.create(persona=direttore_corso, stato=Delega.ATTIVA, tipo=DIRETTORE_CORSO,
+                              inizio=poco_fa(), oggetto_tipo=ctype,
+                              oggetto=corso_base, firmatario=presidente)
+
+        # invito al corso aspirante1 e sostenitore1
+        InvitoCorsoBase.objects.create(persona=aspirante1, corso=corso_base, invitante=direttore_corso)
+        InvitoCorsoBase.objects.create(persona=sostenitore1, corso=corso_base, invitante=direttore_corso)
+
+        # iscritti al corso aspirante2 e sostenitore2
+        partecipazione1 = PartecipazioneCorsoBase.objects.create(persona=aspirante2, corso=corso_base, confermata=True)
+        partecipazione2 = PartecipazioneCorsoBase.objects.create(persona=sostenitore2, corso=corso_base,
+                                                                 confermata=True)
+
+        # autorizzazioni al corso aspirante2 e sostenitore2
+        Autorizzazione.objects.create(richiedente=aspirante2, firmatario=direttore_corso, concessa=True,
+                                      oggetto_tipo=58, oggetto=partecipazione1, progressivo=1,
+                                      destinatario_ruolo="CB-part", destinatario_oggetto=corso_base, necessaria=False)
+        Autorizzazione.objects.create(richiedente=sostenitore2, firmatario=direttore_corso, concessa=True,
+                                      oggetto_tipo=58, oggetto=partecipazione2, progressivo=1,
+                                      destinatario_ruolo="CB-part", destinatario_oggetto=corso_base, necessaria=False)
+
+        # sessione
+        sessione_direttore = self.sessione_utente(persona=direttore_corso)
+        sessione_direttore.visit("%s/aspirante/corso-base/%d/iscritti/" % (self.live_server_url, corso_base.pk))
+        with sessione_direttore.get_iframe(0) as iframe_direttore:
+            link = iframe_direttore.find_link_by_partial_text("Excel: Un foglio per sede").first._element.get_attribute(
+                'href')
+        try:
+            sessione_direttore.visit(link)
+            self.assertFalse(sessione_direttore.is_text_present("Server Error (500)"))
```

Comment:
Per controllare lo stato della richiesta puoi usare [`sessione.status_code`](http://splinter.readthedocs.io/en/latest/http-status-code-and-exception.html).

Piuttosto che controllare la stringa "Server Error (500)", che dipende dal server web, potresti controllare che la richiesta sia andata a buon fine:

```python
self.assertTrue(sessione_direttore.status_code.is_success())
```

Reply:
Questo test è destinato a fallire se l'implementazione della pagina di errore del server di sviluppo di Django cambia (per esempio all'aggiornamento alla versione 2).

Come da discussione in chat, un test più appropriato potrebbe essere qualcosa come
```python
self.assertFalse("<html>" in sessione_direttore.html)
```

Questo perché se la pagina generata è HTML, indipendentemente dal testo nella pagina, sappiamo che non è stato generato un Excel con successo.